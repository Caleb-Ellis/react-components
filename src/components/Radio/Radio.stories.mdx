import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs/blocks";
import { useState } from "react";
import Radio from "./Radio";

<Meta
  title="Radio"
  component={Radio}
  argTypes={{
    label: {
      control: {
        type: "text",
      },
      defaultValue: "Radio",
    },
    disabled: {
      control: {
        type: "boolean",
      },
    },
    required: {
      control: {
        type: "boolean",
      },
    },
    indeterminate: {
      control: {
        type: "boolean",
      },
    },
  }}
/>

export const Template = (args) => <Radio {...args} />;

### Radio

TODO: add description.

### Props

<ArgsTable of={Radio} />{" "}

### Default

<Canvas>
  <Story name="Default">{Template.bind({})}</Story>
</Canvas>

### Disabled

<Canvas>
  <Story name="Disabled">
    <Radio label="Disabled" name="Radio" disabled></Radio>
  </Story>
</Canvas>

### Required

<Canvas>
  <Story name="Required">
    <Radio label="Required" name="Radio" required></Radio>
  </Story>
</Canvas>
