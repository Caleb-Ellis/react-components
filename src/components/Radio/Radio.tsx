import PropTypes from "prop-types";
import React, { HTMLProps } from "react";

import CheckableInput from "../CheckableInput";

type Props = {
  label: string;
} & HTMLProps<HTMLElement>;

const Radio = ({ type, label, ...radioProps }: Props): JSX.Element => {
  return (
    <CheckableInput type="radio" label={label} {...radioProps}></CheckableInput>
  );
};

Radio.propTypes = {
  label: PropTypes.string.isRequired,
};

export default Radio;
