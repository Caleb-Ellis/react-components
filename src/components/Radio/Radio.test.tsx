import { shallow } from "enzyme";
import React from "react";

import Radio from "./Radio";

describe("Radio ", () => {
  it("renders", () => {
    const wrapper = shallow(<Radio label="Test radio"></Radio>);
    expect(wrapper).toMatchSnapshot();
  });
});
