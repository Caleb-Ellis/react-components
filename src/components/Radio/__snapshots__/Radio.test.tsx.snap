// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio  renders 1`] = `
<CheckableInput
  label="Test radio"
  type="radio"
/>
`;
